tests/jballerina-semtype-port-test/src/test/java/io/ballerina/semtype/port/test/SemTypeResolver.java

```diff
@@ -605,12 +610,55 @@ private SemType resolveSingletonType(Object value, TypeKind targetTypeKind) {
 
     private SemType resolveTypeDesc(Context cx, Map<String, BLangNode> mod, BLangTypeDefinition defn, int depth,
                                     BLangTableTypeNode td) {
-        if (td.tableKeySpecifier != null || td.tableKeyTypeConstraint != null) {
-            throw new UnsupportedOperationException("table key constraint not supported yet");
+        SemType memberType = resolveTypeDesc(cx, mod, defn, depth, td.constraint);
+
+        if (td.tableKeySpecifier != null) {
```

Comment:
Given this logic is common for both here as well as BTableType (and more or less this is an internal detail of how we represent table type) shall we move this logic in to TableSubType. Maybe we can have a utility function that return some sort of record accepted by the `tableContainig` method, or overrides to `tableContaining`.

Reply:
Refactored in [b0018c1](https://github.com/ballerina-platform/ballerina-lang/pull/43092/commits/b0018c10a6ab21d86e618299515e39f49fe6ffda)